libsolidity/codegen/LValue.cpp

```diff
@@ -198,6 +198,201 @@ void ImmutableItem::setToZero(SourceLocation const&, bool _removeReference) cons
 	m_context << Instruction::POP;
 }
 
+TransientStorageItem::TransientStorageItem(CompilerContext& _compilerContext, VariableDeclaration const& _declaration):
+	TransientStorageItem(_compilerContext, *_declaration.annotation().type)
+{
+	solAssert(!_declaration.immutable(), "");
+	solAssert(_declaration.referenceLocation() == VariableDeclaration::Location::Transient);
+	auto const& location = m_context.storageLocationOfVariable(_declaration);
+	m_context << location.first << u256(location.second);
+}
+
+TransientStorageItem::TransientStorageItem(CompilerContext& _compilerContext, Type const& _type):
+	LValue(_compilerContext, &_type)
+{
+	if (m_dataType->isValueType())
+	{
+		if (m_dataType->category() != Type::Category::Function)
+			solAssert(m_dataType->storageSize() == m_dataType->sizeOnStack(), "");
+		solAssert(m_dataType->storageSize() == 1, "Invalid transient storage size.");
+	}
+}
+
+void TransientStorageItem::retrieveValue(SourceLocation const&, bool _remove) const
+{
+	// stack: storage_key storage_offset
+	if (!m_dataType->isValueType())
+	{
+		solAssert(m_dataType->sizeOnStack() == 1, "Invalid transient storage ref size.");
+		if (_remove)
+			m_context << Instruction::POP; // remove byte offset
+		else
+			m_context << Instruction::DUP2;
+		return;
+	}
```

Comment:
```suggestion
	solUnimplementedAssert(m_dataType->isValueType(), "Transient storage is only supported for value types.");
```

This is technically a dead branch, since it should hopefully be impossible to create a reference type transient variable declaration (i.e. it would error out during analysis); it should none the less be replaced with an assert.

Reply:
Done.